tmk_core/common/action.c

```diff
@@ -889,9 +938,13 @@ void unregister_code(uint8_t code) {
  */
 void tap_code_delay(uint8_t code, uint16_t delay) {
     register_code(code);
+#if defined(__AVR__)
```

Comment:
Is there a specific reason for this change?

Reply:
Waiting on the Chibios platform is done by de-scheduling the current thread and re-scheduling after the waiting time. Under the hood this also means setting a wakeup timer and reacting to this timer interrupt. In bulk this is rather costly, so I merged this into one continuous waiting block.

Reply:
Im not a fan of this change, as it bleeds more platform knowledge into code code (and from what i can tell, isnt strictly required for the overall aim of this PR).

Personally I would like to see it PRed separately, maybe hiding the implementation difference within the wait_ms AVR code.

Reply:
Reverted, I'm going to open a separate PR once this is merged.